src/java/org/apache/cassandra/db/AbstractBufferClusteringPrefix.java

```diff
@@ -42,10 +42,18 @@ public ByteBuffer[] getBufferArray()
         return getRawValues();
     }
 
-    public ClusteringPrefix<ByteBuffer> minimize()
+    public ClusteringPrefix<ByteBuffer> retainable()
     {
         if (!ByteBufferUtil.canMinimize(values))
             return this;
-        return new BufferClustering(ByteBufferUtil.minimizeBuffers(values));
+
+        ByteBuffer[] minimizedValues = ByteBufferUtil.minimizeBuffers(this.values);
+        if (kind.isBoundary())
+            return accessor().factory().boundary(kind, minimizedValues);
+        if (kind.isBound())
+            return accessor().factory().bound(kind, minimizedValues);
+
+        assert kind() != Kind.STATIC_CLUSTERING;    // not minimizable
+        return accessor().factory().clustering(minimizedValues);
```

Comment:
This is equivalent to `return new BufferClustering(values);`

Reply:
Clarity and robustness. I'd prefer calling `ByteBufferObjectFactory`'s methods directly, but that class is not exposed.